arraycontext/__init__.py

```diff
@@ -47,12 +47,16 @@
         rec_multimap_array_container,
         mapped_over_array_containers,
         multimapped_over_array_containers,
+        keyed_map_array_container,
+        rec_keyed_map_array_container,
```

Comment:
Kind of don't like these enough to have them exported to the main namespace.

Reply:
I thought they were helpful routine, but I guess the vagueness of key-spec is a show-stopper: 0980f4b.